PLATFORMS.md

```diff
@@ -0,0 +1,63 @@
+
+Several parts of this policy require providing target-specific documentation. Such documentation should typically appear in a subdirectory of the platform-support section of the `liboqs` manual, with a link from the target's entry in platform support.
+
+Note that a target must have already received approval for the next lower tier, and spent a reasonable amount of time at that tier, before making a proposal for promotion to the next higher tier; this is true even if a target meets the requirements for several tiers at once. This policy leaves the precise interpretation of "reasonable amount of time" up to the approving team; the team may scale the amount of time required based on their confidence in the target and its demonstrated track record at its current tier. At a minimum, multiple stable releases of `liboqs` should typically occur between promotions of a target.
+
+The availability or tier of a target in stable `liboqs` is not a hard stability guarantee about the future availability or tier of that target. Higher-level target tiers are an increasing commitment to the support of a target, and we will take that commitment and potential disruptions into account when evaluating the potential demotion or removal of a target that has been part of a stable release. The promotion or demotion of a target will not generally affect existing stable releases, only current development and future releases.
+
+In this policy, the words "must" and "must not" specify absolute requirements that a target must meet to qualify for a tier. The words "should" and "should not" specify requirements that apply in almost all cases, but for which the approving teams may grant an exception for good reason. The word "may" indicates something entirely optional, and does not indicate guidance or recommendations. This language is based on [IETF RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Platforms supported
+
+### Tier 1
+
+- x86_64/amd64/x64 for Ubuntu Linux (Focal) and MacOS (XCode 14)
+- aarch64 for Ubuntu (Focal)
+- armhf/ARM7 and aarch64 emulation on Ubuntu
+- ppc64le for Ubuntu (Focal)
+- s390x for Ubuntu (Focal)
+
+### Tier 2
+
+- x86_64/amd64/x64 for Windows 2022
+- armeabi-v7a, arm64-v8a, x86, x86_64 for Android
+- OS64, TVOS for Apple Mobile
+
+### Tier 3
+
+- x86 for Windows
+- aarch64 for MacOS
```

Comment:
aarch64 is clearly the intended future of the macOS platform, so we would definitely want to get this up to tier 1.  Is it not higher right now because we don't have any CI on this platform?

Reply:
>  Is it not higher right now because we don't have any CI on this platform

Correct

Reply:
Github now has M1 runners I believe. I learnt this when someone was complaining about how expensive they are, though, and I'm not sure if they are available for OSS yet.

Reply:
> I'm not sure if they are available for OSS yet.

They are not: See [documentation](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners#machine-specs-for-larger-runners) and [worked example](https://github.com/open-quantum-safe/oqs-provider/actions/runs/7075061285/job/19256776098).

Reply:
For completeness: Error message:

```
The job was not started because recent account payments have failed or your spending limit needs to be increased. Please check the 'Billing & plans' section in your settings.
```

Reply:
I agree that aarch64/Mac is correctly classified as "tier 3" but ideally would be higher. Perhaps we could look into using our M1 profiling machine as a self-hosted runner? I don't think we would want the strain of running tests on it as often as on the GitHub runners, but a scheduled daily or weekly job would be more palatable.

Reply:
Agreed. Also thought the same. But wouldn't this effort be a waste of time? The LF companies will pay hundreds of thousands -- why not use (a tiny fraction) of this money to pay for M1 runners and save us this rather tedious work? It surely would be a better use of the money than for CO2-gobbling trips of some LF executives to PQ "workshops" :)

Reply:
I've created issue #1630 to track this, and will close this thread as resolved.